flutter/ios/Classes/SwiftSentryFlutterPlugin.swift

```diff
@@ -3,27 +3,118 @@ import Sentry
+            options.dist = arguments["dist"] as? String
+            options.integrations = arguments["integrations"] as? [String] ?? []
+            options.maxBreadcrumbs = arguments["maxBreadcrumbs"] as? UInt ?? 100
+
+            /*
+             missing fields : enableAutoNativeBreadcrumbs, diagnostic level,
+             enableNativeCrashHandling,platform, web, packages,anrTimeoutIntervalMillis
+             diagnosticLevel, cacheDirSize
+             */
+
+            options.beforeSend = { event in
+                self.setEventOriginTag(event: event)
+                /// TODO ? addPackages, removeThreadsIfNotAndroid
+                return event
+            }
+            
+            self.options = options
+        }
+
+        result("iOS initNativeSdk" )
+    }
+
+    private func setEventOriginTag(event: Event){
+        guard let sdk = event.sdk else { return  };
+        if self.isValidSdk(sdk:sdk){
+
+            switch sdk["name"] as! String{
+            case "sentry.dart.flutter":
+                setEventEnvironmentTag(event:event,origin:"flutter",environment:"dart")
+            case "sentry.cocoa":
+                setEventEnvironmentTag(event:event,origin: "flutter",environment: "dart")
+            case "sentry.native":
+                setEventEnvironmentTag(event:event,origin: "flutter",environment: "dart")
+            default:
+                return
+            }
+        }
+    }
+
+
+    private func setEventEnvironmentTag(event: Event, origin: String = "ios", environment: String) {
+        event.tags?["event.origin"] = origin
+        event.tags?["event.environment"] = environment
+    }
+
+    private func isValidSdk( sdk: [String: Any]) -> Bool{
+        return (sdk["name"] != nil && !(sdk["name"] as! String).isEmpty)
+    }
+
+    private func captureEnvelope(_ call: FlutterMethodCall, result: @escaping FlutterResult){
+        guard let arguments = call.arguments as? [Any],
+              !arguments.isEmpty,
+              let event = arguments.first as? String else {
+            result(FlutterError(code: "2", message: "Envelope is null or empty", details: nil) )
+            return
+        }
+        
+        guard writeEnvelope(envelope: event) == true else {
+            result(FlutterError(code: "3", message: "SentryOptions or outboxPath are null or empty", details: nil) )
+            return
+        }
+        result("")
+    }
+    
+    private func writeEnvelope(envelope: String) -> Bool{
+        if let dir = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask).first {
+            let filePath = dir.appendingPathComponent(UUID().uuidString)
+            do{
+                try envelope.write(to: filePath, atomically: false, encoding: .utf8)
+                return true
+            } catch {
+                print("writeEnvelope fail")
+                // logger ?
+            }
+        }
+        return false;
+    }
```

Comment:
should not be 'writeEnvelope', we call `captureEnvelope` on iOS as explained, it's similar to the RN SDK

Reply:
✅

Reply:
✅